src/wechaty_plugin_contrib/contrib/chat_history_plugin/plugin.py

```diff
@@ -0,0 +1,110 @@
+"""Chat history plugin"""
+import os
+from typing import Optional, List, Any
+from dataclasses import dataclass, field
+from wechaty_puppet import MessageType, FileBox
+from wechaty import Wechaty, Message, get_logger
+from wechaty.plugin import WechatyPlugin, WechatyPluginOptions
+from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
+from sqlalchemy.orm import sessionmaker, declarative_base
+from sqlalchemy import (
+    Column,
+    Integer,
+    VARCHAR,
+    Text
+)
+
+logger = get_logger('ChatHistoryPlugin')
+
+SUPPORTED_MESSAGE_FILE_TYPES: List[MessageType] = [
+    MessageType.MESSAGE_TYPE_ATTACHMENT,
+    MessageType.MESSAGE_TYPE_IMAGE,
+    MessageType.MESSAGE_TYPE_EMOTICON,
+    MessageType.MESSAGE_TYPE_VIDEO,
+    MessageType.MESSAGE_TYPE_AUDIO
+]
+
+Base: Any = declarative_base()
+
+
+class ChatHistory(Base):
+    """ChatHistory"""
+    __tablename__ = 'ChatHistory'
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    msg_id = Column(Integer, default=None)
+    filename = Column(Text, default=None)
+    text = Column(Text, default=None)
+    timestamp = Column(Integer, default=None)
+    type = Column(Integer, default=None)
+    from_id = Column(VARCHAR(50), default=None)
+    room_id = Column(VARCHAR(50), default=None)
+    to_id = Column(VARCHAR(50), default=None)
+    mention_ids = Column(Text, default=None)
+
+
+@dataclass
+class ChatHistoryPluginOptions(WechatyPluginOptions):
+    """
+    chat history plugin options
+    """
+    chat_history_path: str = field(default_factory=str)
+    chat_history_database: str = field(default_factory=str)
+
+
+class ChatHistoryPlugin(WechatyPlugin):
+    """chat history plugin"""
+
+    def __init__(self, options: Optional[ChatHistoryPluginOptions] = None):
+        super().__init__(options)
+        if options is None:
+            options = ChatHistoryPluginOptions()
+        if not options.chat_history_path:
+            self.chat_history_path = os.path.join(os.getcwd(), 'chathistory')
```

Comment:
If there is no `chathistory` dir in developer current working directory, it will throw FileNotFound error, so you should init it here. Do you think so ??

Btw, i suggest `.wechaty/chathistory` dirname other than `chathistory`. How do you think about it ?

Reply:
- The folder will be initialized in line 64~65:
```python
if not os.path.exists(self.chat_history_path):
    os.makedirs(self.chat_history_path)
```
- Sure. Change to `.wechaty/chathistory` maybe more meaningful.